src/main/java/com/onlinebookstore/onlinebookstore/model/Category.java

```diff
@@ -0,0 +1,40 @@
+package com.onlinebookstore.onlinebookstore.model;
+
+import jakarta.persistence.Column;
+import jakarta.persistence.Entity;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import jakarta.persistence.ManyToMany;
+import jakarta.persistence.Table;
+import jakarta.validation.constraints.NotBlank;
+import java.util.HashSet;
+import java.util.Set;
+import lombok.Getter;
+import lombok.Setter;
+import org.hibernate.annotations.SQLDelete;
+import org.hibernate.annotations.SQLRestriction;
+
+@Entity
+@Table(name = "categories")
+@Getter
+@Setter
+@SQLDelete(sql = "UPDATE categories SET is_deleted = true WHERE id=?")
+@SQLRestriction("is_deleted = false")
+public class Category {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+
+    @NotBlank
+    @Column(unique = true)
+    private String name;
+
+    private String description;
+
+    @Column(name = "is_deleted", nullable = false)
+    private boolean isDeleted = false;
+    @ManyToMany(mappedBy = "categories")
+    private Set<Book> books = new HashSet<>();

```

Comment:
Are you sure that relation should be bi-directional?

Reply:
Well I thought that one book could have many categories like - fantasy, adventure \ sci-fi advanture \ romance, adventure